### PR TITLE
change Account to interface

### PR DESCRIPTION
**Pull request by dustinxie**
Account is basically same as keypair.PrivateKey, change it to an interface and simplified code a lot

### CODE REVIEW

**Comment by ququzone** [comment]
I will refactor this to new contract execution issue. #41

File: account/accounts_test.go [hunk 1]
```diff
@@ -13,18 +13,28 @@ import (
...
-	assert.NotEmpty(t, act.PublicKey)
-	assert.NotEmpty(t, act.Address)
+	assert.NoError(err)
+	assert.NotEmpty(act.PrivateKey())
+	assert.NotEmpty(act.PublicKey())
+	assert.NotEmpty(act.Address)
+
+	b, err := acts.GetAccount(act.Address())
+	assert.NoError(err)
+	assert.Equal(act, b)
```
Comment (dustinxie) [inline account/accounts_test.go]: verify the account saved into map == origin account created

File: action/action.go [hunk 2]
```diff
@@ -25,41 +24,21 @@ type (
...
-		return nil, err
-	}
-	h := hash.Hash256b(msg)
-	return h[:], nil
-}
-
 // Sign signs the ActionCore
-func (ac *IotexActionCore) Sign(sk keypair.PrivateKey) (*IotexAction, error) {
-	h, err := ac.Hash()
+func (ac *IotexActionCore) Sign(act account.Account) (*IotexAction, error) {
```
Comment (dustinxie) [inline action/action.go]: pass in Account to sign, no need to expose the PrivateKey, this also allows us to remove the 2 Hash() functions

File: iotx/iotx_test.go [hunk 3]
```diff
@@ -48,10 +46,9 @@ func TestDeployContract(t *testing.T) {
 	require.NoError(err)
 	acc, err := iotx.Accounts.PrivateKeyToAccount(accountPrivateKey)
 	require.NoError(err)
-	require.EqualValues(acc.Address, accountAddress)
```
Comment (dustinxie) [inline iotx/iotx_test.go]: this check was done in account_test.go, remove it here

**Comment by ququzone** [comment]
work for #44

**Pull request merged**